core/services/ocrcommon/transmitter.go

```diff
@@ -57,9 +71,22 @@ func (t *transmitter) FromAddress() common.Address {
 	return t.effectiveTransmitterAddress
 }
 
+func (t *transmitter) FromAddressForTransaction() common.Address {
+	// Use Round-Robin to select the next fromAddress.
+	nextFromAddress := t.fromAddresses[t.nextFromAddressIndex]
+	t.nextFromAddressIndex++
+	if t.nextFromAddressIndex >= len(t.fromAddresses) {
+		t.nextFromAddressIndex = 0
+	}
```

Comment:
Minor: I think you can simplify this and make it clearer that it's a round-robin by using `t.nextFromAddressIndex = (t.nextFromAddressIndex + 1) % len(t.fromAddresses)`. Either way this method could use a unit test.